### DISCUSSION TITLE
after commas, now white spaces get removed from wildcards...

### DISCUSSION THREAD

**Discussion by rtaskf** [adieyal/sd-dynamic-prompts#694] [General]
So, after the comma issue was fixed (https://github.com/adieyal/sd-dynamic-prompts/discussions/672), now it's the white spaces that get removed from the beginning/end of wildcards. This has broken all my files again. Can you please fix that and let the user decide where they want commas and white spaces? I don't see the need to further pre-process the strings. It only makes the system more restrictive and breaks existing "libraries" every time.

I can no longer use something like "{|oil |water }painting{| on the wall}", as it becomes "oilpaintingon the wall" without the "padding".

**Comment by rtaskf** [comment]
Will this be addressed or not??

**Reply by akx** [reply]
I'm sorry you're having these problems, but please do remember this project is being maintained by approximately two people in their spare time, and demanding an issue be fixed is unlikely to expedite the process.

**Reply by rtaskf** [reply]
Last time, not long ago, it took only hours to get a response and the issue was fixed within 2 days or so. Naturally, I wasn't expecting not to get any response at all in over a week this time. It would be nice to know if this is even considered a bug or a "feature". Maybe they think that, for some reason, they need to further process the "raw" strings somehow, which I absolutely disagree with.

**Reply by akx** [reply]
I'm one of the two people maintaining this project, and I was the one who fixed the issue you last reported, so chances are I'm part of the "they" you're referring to.

It may well be this is a bug or unintentional regression, but no one has had the time to investigate and/or fix yet.

As for

> which I absolutely disagree with.

, the great thing about open source projects is you can use and modify them for your own purposes for free!

**Comment by akx** [comment]
Okay, @rtaskf, I dug into this a bit now that I had the time.

This is pretty much expected behavior in `dynamicprompts`, and has been practically forever (I didn't dig further back; please do if you like): 

```
0.20.0 ['oil painting', 'oil paintingon the wall', 'painting', 'paintingon the wall', 'water painting', 'water paintingon the wall']
0.21.0 ['oil painting', 'oil paintingon the wall', 'painting', 'paintingon the wall', 'water painting', 'water paintingon the wall']
0.22.0 ['oil painting', 'oil paintingon the wall', 'painting', 'paintingon the wall', 'water painting', 'water paintingon the wall']
0.23.0 ['oil painting', 'oil paintingon the wall', 'painting', 'paintingon the wall', 'water painting', 'water paintingon the wall']
0.24.0 ['oil painting', 'oil paintingon the wall', 'painting', 'paintingon the wall', 'water painting', 'water paintingon the wall']
0.25.0 ['oil painting', 'oil paintingon the wall', 'painting', 'paintingon the wall', 'water painting', 'water paintingon the wall']
0.26.0 ['oil painting', 'oil paintingon the wall', 'painting', 'paintingon the wall', 'water painting', 'water paintingon the wall']
0.27.0 ['oil painting', 'oil paintingon the wall', 'painting', 'paintingon the wall', 'water painting', 'water paintingon the wall']
0.28.0 ['oil painting', 'oil paintingon the wall', 'painting', 'paintingon the wall', 'water painting', 'water paintingon the wall']
0.29.0 ['oil painting', 'oil paintingon the wall', 'painting', 'paintingon the wall', 'water painting', 'water paintingon the wall']
0.30.1 ['oil painting', 'oil paintingon the wall', 'painting', 'paintingon the wall', 'water painting', 'water paintingon the wall']
0.30.4 ['oil painting', 'oil paintingon the wall', 'painting', 'paintingon the wall', 'water painting', 'water paintingon the wall']
```

(If you can point out a version that has acted differently, that'd be very interesting to see!)

We strip leading whitespace out of variant members, for one because it makes writing variants out as multiple lines much cleaner.

The easy workaround is turning your

```
{|oil |water }painting{| on the wall}
```
into
```
{|oil |water }painting {|on the wall}
```
since we naturally never strip any whitespace out from literals.

If you like, you can turn "Ignore whitespace in prompts" on in the extension settings so there'll be no stray spaces left, but I don't think they affect the end result in any way.

Hope this helps, and happy holidays!

**Reply by rtaskf** [reply]
"We strip leading whitespace out of variant members, for one because it makes writing variants out as multiple lines much cleaner."

That's funny, because I'm using one yaml file with multi-line options (all my other files are simple txt), but that file always produces newlines in the final prompt anyway when used.
In any case, wouldn't it make more sense to remove the newlines and keep the white spaces?...

Happy holidays!
